projects/core/src/site-theme/facade/site-theme.service.ts

```diff
@@ -48,26 +74,31 @@ export class SiteThemeService {
   /**
    * Sets the active theme className.
    */
-  // TODO(#8153): If we move this._isInitialized into an RxJS stream, it may create a race condition between SiteThemePersistenceService:onRead and SiteThemeInitializer:this.setFallbackValue().
   setActive(className: string): void {
-    this.isValidTheme(className)
-      .pipe(filter(Boolean), take(1))
-      .subscribe(() => {
-        this.store.dispatch(new SiteThemeActions.SetActiveSiteTheme(className));
+    this.store
+      .pipe(select(SiteThemeSelectors.getActiveSiteTheme), take(1))
+      .subscribe((activeTheme: string | null) => {
+        if (activeTheme !== className) {
+          this.store.dispatch(
+            new SiteThemeActions.SetActiveSiteTheme(className)
+          );
+        }
```

Comment:
```suggestion
        if (this.isValidTheme(className) && activeTheme !== className) {
          this.store.dispatch(
            new SiteThemeActions.SetActiveSiteTheme(className)
          );
        }
```
`major:` I think the check if value is valid is still required before we call an action to set new active theme. Without this, we will still encounter the issue we discussed where action is can be triggered with the invalid value from local storage.

Reply:
oh, good catch. thanks. done